posydon/popsyn/binarypopulation.py

```diff
@@ -830,46 +837,94 @@ def draw_initial_binary(self, **kwargs):
         output = self.draw_initial_samples(**sampler_kwargs)
 
         default_index = output['binary_index'].item()
+        #Eirini's comments:
         # Randomly generated variables
-        formation_time = output['time'].item()
-        separation = output['separation'].item()
-        orbital_period = output['orbital_period'].item()
-        eccentricity = output['eccentricity'].item()
-        m1 = output['S1_mass'].item()
-        m2 = output['S2_mass'].item()
-        Z_div_Zsun = kwargs.get('metallicity', 1.)
-        zams_table = {1.: 2.703e-01,
-                      0.1: 2.511e-01,
-                      0.01: 2.492e-01,
-                      0.001: 2.49e-01,
-                      0.0001: 2.49e-01}
-        Y = zams_table[Z_div_Zsun]
-        Z = Z_div_Zsun*Zsun
-        X = 1. - Z - Y
-
-        binary_params = dict(
-            index=kwargs.get('index', default_index),
-            time=formation_time,
-            state="detached",
-            event="ZAMS",
-            separation=separation,
-            orbital_period=orbital_period,
-            eccentricity=eccentricity,
-        )
-        star1_params = dict(
-            mass=m1,
-            state="H-rich_Core_H_burning",
-            metallicity=Z,
-            center_h1=X,
-            center_he4=Y,
-        )
-        star2_params = dict(
-            mass=m2,
-            state="H-rich_Core_H_burning",
-            metallicity=Z,
-            center_h1=X,
-            center_he4=Y,
-        )
+
+
+        if random.random()<self.binary_fraction:
```

Comment:
I would suggest a different way to generate binaries given a binary fraction. For this PR maybe we can use it and change later but I would like to have a more detailed discussion. And another separate note - using a random process like this which is not using the global seed with numpy will not be reproducible.

Reply:
And to suggest an actual fix if you want to keep it as is:
```python
if self.RNG.uniform() < self.binary_fraction:
```